src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java

```diff
@@ -64,97 +64,68 @@ public ActiveSiteAlignmentResponse alignActiveSites(ActiveSiteAlignmentRequest a
         Page<Motif> motifs = motifService.queryByEcNumber(motifEcNumberFilter, pageNumber);
 
         log.info(String.format("Aligning active sites of %d PDB entries with %d motifs & %d custom motifs.",
-                sourceStructures.size(), motifs.getTotalElements(), customMotifs.size()));
+                sourceStructures.size(), motifs.getTotalElements(), customMotifFileList.size()));
 
         // Align structures with motifs from the database
         while (motifs.hasContent()) {
-            motifs.stream()
-                    .parallel()
-                    .forEach(motif -> {
-                        Structure motifStructure = ProteinUtils.queryPdb(motif.getPdbId());
-                        alignActiveSiteStructureList(results, motif, motifStructure, sourceStructures, precision);
-                    });
+            for (Structure structure : sourceStructures) {
+                results.get(structure.getPDBCode())
+                        .addAll(motifs.stream().parallel()
+                                .map(motif -> alignActiveSites(structure, motif, ProteinUtils.queryPdb(motif.getPdbId()), precision))
+                                .filter(Objects::nonNull)
+                                .collect(Collectors.toList()));
+            }
             pageNumber++;
             motifs = motifService.queryByEcNumber(motifEcNumberFilter, pageNumber);
         }
 
         // Align structures with custom uploaded motifs
-        customMotifs.forEach((customMotif, customMotifStructure) -> {
-            alignActiveSiteStructureList(results, customMotif, customMotifStructure, sourceStructures, precision);
-        });
-
-        results.forEach((pdbId, alignments) -> results.replace(pdbId, alignments.stream()
-                .filter(Objects::nonNull)
-                .collect(Collectors.toList())));
+        if (customMotifFileList.size() > 0) {
```

Comment:
By including this if we eliminate a for loop across a large number of entities (we would iterate through and do nothing with the interior stream) when we do not have to.